nuxeo-management-rest-api/src/test/java/org/nuxeo/rest/management/BaseTest.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Nour Al Kotob
+ */
+
+package org.nuxeo.rest.management;
+
+import javax.inject.Inject;
+
+import org.junit.Before;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.core.api.impl.blob.JSONBlob;
+import org.nuxeo.jaxrs.test.CloseableClientResponse;
+import org.nuxeo.jaxrs.test.HttpClientTestRule;
+import org.nuxeo.jaxrs.test.HttpClientTestRule.Builder;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.ServletContainerFeature;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features(ManagementFeature.class)
+public class BaseTest {
+
+    protected static final String ADMINISTRATOR = "Administrator";
+
+    @Inject
+    protected ServletContainerFeature servletContainerFeature;
+
+    protected ObjectMapper mapper;
+
+    @Before
+    public void setup() {
+        mapper = new ObjectMapper();
+    }
+
+    protected CloseableClientResponse get(String path) {
+        String url = "http://localhost:" + servletContainerFeature.getPort() + "/";
+        Builder builder = new HttpClientTestRule.Builder().accept(JSONBlob.APPLICATION_JSON)
+                                                          .credentials(ADMINISTRATOR, ADMINISTRATOR)
+                                                          .url(url);
+        HttpClientTestRule response = builder.build();
+        response.starting();
+        CloseableClientResponse result = response.get(path);
+        response.finished();
+        return result;
+    }
+}
```

Comment:
thanks, :)
its seem good, but i think we can make some improvements (nice to have) like 
` protected ObjectMapper mapper = new ObjectMapper();


    protected CloseableClientResponse get(String path) {
        Builder builder = new HttpClientTestRule.Builder().accept(JSONBlob.APPLICATION_JSON)
                .credentials(ADMINISTRATOR, ADMINISTRATOR)
                .url(String.format("http://localhost:%s/", servletContainerFeature.getPort()));

        HttpClientTestRule response = builder.build();
        response.starting();
        CloseableClientResponse result = response.get(path);
        response.finished();
        return result;
    }`
- 1) i dont think the `before` method is required, we can we create the object mapper on in line way
-2) can replace the url var creation by String#format

for the status codes checking, i think use the `org.apache.http.HttpStatus` and 
`assertEquals(HttpStatus.SC_OK, response.getStatus());` with an static import on code 
assertEquals(SC_OK, response.getStatus());

wdyt,

Reply:
The @before executes before each test. That's why i left the setup method.
What do you think about it ?
About status code ok for me !

Reply:
yep i agree for `@before` method is before each test, but the `ObjectMapper` is thread safe, is just used to read tthe json response, stateless. a unique instance is is fully enough, we can have another point view from @ataillefer

Reply:
- Yes about ObjectMapper initialized inline => no `@Before` needed for this.
- Sure for String#format instead of +, though it's really a detail here in the context of unit tests.
Yet, I'm personally still in favor of defining some intermediate variables such as `url` as it's more readable and eases debugging.
I'd write:
```
String url = String.format("http://localhost:%s/", servletContainerFeature.getPort());
call(url)
```

Reply:
About the status codes:
- Usually we use `javax.ws.rs.core.Response.Status` or `javax.servlet.http.HttpServletResponse`, see https://github.com/nuxeo/nuxeo/blob/master/addons/nuxeo-wopi/nuxeo-wopi/src/main/java/org/nuxeo/wopi/jaxrs/FilesEndpoint.java#L23
- Anyway, in practice, I find it clearer to have the explicit integer rather than a constant for the expected value. My opinion...

Reply:
In fact, you should really take example on how the `HttpClientTestRule` is used, only one test AFAIK: https://github.com/nuxeo/nuxeo/blob/10.10/nuxeo-features/nuxeo-platform-forms-layout-export/src/test/java/org/nuxeo/ecm/platform/forms/layout/export/TestLayoutResource.java

So you'll need à `@Before` to initialize the rule (protected member) and start it, an `@After` to finish it, and no more need of the `get` method, just call `httpClientRule.get` directly!

Reply:
agree for using the `HttpServletResponse `. Use of constants, give us a short descritpion of the status, mainly where we expect a status that we don't use often. It's a nice to have and not mandatory. I think the idea is to be consistent with the other code of the project.

Reply:
Sure, whatever works!

Reply:
Implemented all that. About the status code I used the HttpServletResponse. But I like numbers better. I actually had to look up for which one of the constants would send a 500